pkg/reconciler/v1alpha1/eventlistener/eventlistener.go

```diff
@@ -66,8 +77,119 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
+			// If our EventListener is deleted, then its Deployment should be as well
+			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
+			Labels:          labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "event-listener",
+							Image: *elImage,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: int32(8082),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	oldDeployment, err := c.KubeClientSet.AppsV1().Deployments(el.Namespace).Get(el.Name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			c.Logger.Errorf("Error getting Deployments: %s", err)
+			return err
+		}
+
+		// Create the EventListener Deployment
+		_, err = c.KubeClientSet.AppsV1().Deployments(el.Namespace).Create(deployment)
+		if err != nil {
+			c.Logger.Errorf("Error creating EventListener Deployment: %s", err)
+			return err
+		}
+	} else if !reflect.DeepEqual(oldDeployment, deployment) {
+		// Update the EventListener Deployment
+		_, err = c.KubeClientSet.AppsV1().Deployments(el.Namespace).Update(deployment)
+		if err != nil {
+			c.Logger.Errorf("Error updating EventListener Deployment: %s", err)
+			return err
+		}
+	}
+
+	// TODO: This is an example Service that we will probably want to modify in the future
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			// Create the EventListener's Service in the same Namespace as where the
+			// EventListener was created
+			Namespace: el.Namespace,
+			// Give the Service the same name as the EventListener
+			Name: el.Name,
+			// If our EventListener is deleted, then its Service should be as well
+			OwnerReferences: []metav1.OwnerReference{*el.GetOwnerReference()},
+			Labels:          labels,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": el.Name},
+			Type:     corev1.ServiceTypeLoadBalancer,
+			Ports: []corev1.ServicePort{
+				{
+					Protocol: corev1.ProtocolTCP,
+					Port:     int32(8082),
+				},
+			},
+		},
+	}
+	oldService, err := c.KubeClientSet.CoreV1().Services(el.Namespace).Get(el.Name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			c.Logger.Errorf("Error getting Services: %s", err)
+			return err
+		}
+
+		// Create the EventListener Service
+		_, err = c.KubeClientSet.CoreV1().Services(el.Namespace).Create(service)
+		if err != nil {
+			c.Logger.Errorf("Error creating EventListener Service: %s", err)
+			return err
+		}
+	} else if !reflect.DeepEqual(oldService, service) {
+		// Update the EventListener Service
+		_, err = c.KubeClientSet.CoreV1().Services(el.Namespace).Update(service)
+		if err != nil {
+			c.Logger.Errorf("Error updating EventListener Service: %s", err)
+			return err
+		}
+	}
```

Comment:
in future iterations i think we should start moving this logic out of hte reconciler as much as possible into a separate package or two :D (with unit tests 😇 )

(as few methods and as little logic in the reconciler as possible i always say!)

we should also introduce some reconciler level tests ("integration" tests) like https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

Reply:
Good idea to move the logic into a separate package 👍 
Also, I'm already ahead of you on the reconciler tests, doing the work in my [el-reconciler-test](https://github.com/ncskier/triggers/tree/el-reconciler-test) branch